ONSdigital/dp-code-list-api
# use dp-graph with sorted codes

### What

- Use the latest version of dp-graph to obtain the codes in the right order.
- Used `CountCodes` method to obtain totalCount in GetCodes, so that we only call `GetCodes` if needed.
- Removed sort, we have to use the order as returned by Neptune.
- Changed and added unit tests accordingly

### How to review

- Make sure code changes make sense
- Make sure unit tests pass
- (optional) I manually tested these changes with Postman:
  - force use neptune graph:
```
export GRAPH_DRIVER_TYPE=neptune
export GRAPH_ADDR=ws://localhost:8182/gremlin
```
  - run `make debug`
  - tunel to dev-test neptune cluster : `dp ssh  develop web 1 -p 8182:develop-neptune-dev-test-cluster.cluster-cpviojtnaxsj.eu-west-1.neptune.amazonaws.com:8182`
  -  get codes with order: `GET http://localhost:22400/code-lists/mmm/editions/one-off/codes`
    - check that the codes are returned in the natural order (jan, feb, ..., dec)
  -  get codes without order: `GET http://localhost:22400/code-lists/adzuna-jobs-category/editions/one-off/codes`
    - check that the codes are returned in alphabetical order

### Who can review

Anyone
